transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -383,6 +384,43 @@ protected final int doWriteBytes(ChannelOutboundBuffer in, ByteBuf buf) throws E
         return WRITE_STATUS_SNDBUF_FULL;
     }
 
+    /**
+     * Write bytes to the socket, with or without a remote address.
+     * Used for datagram and TCP client fast open writes.
+     */
+    protected final long doWriteOrSendBytes(ByteBuf data, InetSocketAddress remoteAddress, boolean fastOpen)
```

Comment:
moved this up. it is now technically a 'stateless' write/send, instead of just being for datagrams.